pkg/packet/bgp/sr_policy.go

```diff
@@ -676,19 +676,86 @@ func (s *SegmentTypeA) MarshalJSON() ([]byte, error) {
+	END_WITH_PSP_USP      SRBehavior = SRBehavior(api.SRv6Behavior_END_WITH_PSP_USP)
+	ENDX                  SRBehavior = SRBehavior(api.SRv6Behavior_ENDX)
+	ENDX_WITH_PSP         SRBehavior = SRBehavior(api.SRv6Behavior_ENDX_WITH_PSP)
+	ENDX_WITH_USP         SRBehavior = SRBehavior(api.SRv6Behavior_ENDX_WITH_USP)
+	ENDX_WITH_PSP_USP     SRBehavior = SRBehavior(api.SRv6Behavior_ENDX_WITH_PSP_USP)
+	ENDT                  SRBehavior = SRBehavior(api.SRv6Behavior_ENDT)
+	ENDT_WITH_PSP         SRBehavior = SRBehavior(api.SRv6Behavior_ENDT_WITH_PSP)
+	ENDT_WITH_USP         SRBehavior = SRBehavior(api.SRv6Behavior_ENDT_WITH_USP)
+	ENDT_WITH_PSP_USP     SRBehavior = SRBehavior(api.SRv6Behavior_ENDT_WITH_PSP_USP)
+	END_B6_ENCAPS         SRBehavior = SRBehavior(api.SRv6Behavior_END_B6_ENCAPS)
+	END_BM                SRBehavior = SRBehavior(api.SRv6Behavior_END_BM)
+	END_DX6               SRBehavior = SRBehavior(api.SRv6Behavior_END_DX6)
+	END_DX4               SRBehavior = SRBehavior(api.SRv6Behavior_END_DX4)
+	END_DT6               SRBehavior = SRBehavior(api.SRv6Behavior_END_DT6)
+	END_DT4               SRBehavior = SRBehavior(api.SRv6Behavior_END_DT4)
+	END_DT46              SRBehavior = SRBehavior(api.SRv6Behavior_END_DT46)
+	END_DX2               SRBehavior = SRBehavior(api.SRv6Behavior_END_DX2)
+	END_DX2V              SRBehavior = SRBehavior(api.SRv6Behavior_END_DX2V)
+	END_DT2U              SRBehavior = SRBehavior(api.SRv6Behavior_RESERVED)
+	END_DT2M              SRBehavior = SRBehavior(api.SRv6Behavior_END_DT2M)
+	END_B6_ENCAPS_Red     SRBehavior = SRBehavior(api.SRv6Behavior_END_B6_ENCAPS_Red)
+	END_WITH_USD          SRBehavior = SRBehavior(api.SRv6Behavior_END_WITH_USD)
+	END_WITH_PSP_USD      SRBehavior = SRBehavior(api.SRv6Behavior_END_WITH_PSP_USD)
+	END_WITH_USP_USD      SRBehavior = SRBehavior(api.SRv6Behavior_END_WITH_USP_USD)
+	END_WITH_PSP_USP_USD  SRBehavior = SRBehavior(api.SRv6Behavior_END_WITH_PSP_USP_USD)
+	ENDX_WITH_USD         SRBehavior = SRBehavior(api.SRv6Behavior_ENDX_WITH_USD)
+	ENDX_WITH_PSP_USD     SRBehavior = SRBehavior(api.SRv6Behavior_ENDX_WITH_PSP_USD)
+	ENDX_WITH_USP_USD     SRBehavior = SRBehavior(api.SRv6Behavior_ENDX_WITH_USP_USD)
+	ENDX_WITH_PSP_USP_USD SRBehavior = SRBehavior(api.SRv6Behavior_ENDX_WITH_PSP_USP_USD)
+	ENDT_WITH_USD         SRBehavior = SRBehavior(api.SRv6Behavior_ENDT_WITH_USD)
+	ENDT_WITH_PSP_USD     SRBehavior = SRBehavior(api.SRv6Behavior_ENDT_WITH_PSP_USD)
+	ENDT_WITH_USP_USD     SRBehavior = SRBehavior(api.SRv6Behavior_ENDT_WITH_USP_USD)
+	ENDT_WITH_PSP_USP_USD SRBehavior = SRBehavior(api.SRv6Behavior_ENDT_WITH_PSP_USP_USD)
+)
+
 type SRv6EndpointBehaviorStructure struct {
-	Behavior api.SRv6Behavior
+	Behavior SRBehavior
 	BlockLen uint8
 	NodeLen  uint8
 	FuncLen  uint8
 	ArgLen   uint8
 }
 
+func (s *SRv6EndpointBehaviorStructure) DecodeFromBytes(data []byte) error {
+	behavior := binary.BigEndian.Uint16(data[0:2])
```

Comment:
if length of data is less than 2 you will get panic. Please validate data length before doing any manipulations

Reply:
done, thanks